tools/sctool/src/main/java/com/horizen/CommandProcessor.java

```diff
@@ -204,12 +329,20 @@ private void processGenesisInfo(JsonNode json) {
             SidechainProofsCompanion sidechainProofsCompanion = new SidechainProofsCompanion(new HashMap<>());
             SidechainTransactionsCompanion sidechainTransactionsCompanion = new SidechainTransactionsCompanion(new HashMap<>(), sidechainBoxesDataCompanion, sidechainProofsCompanion);
 
-            // TODO: hardcoded
-            ForgerBox forgerBox = SidechainCreation.getHardcodedGenesisForgerBox();
-            PrivateKey25519 forgerSecret = SidechainCreation.genesisSecret;
+            //Find Sidechain creation information
+            SidechainCreation sidechainCreation = null;
+            for (SidechainRelatedMainchainOutput output : mcRef.data().sidechainRelatedAggregatedTransaction().get().mc2scTransactionsOutputs()) {
+                if (output instanceof SidechainCreation) {
+                    sidechainCreation =  (SidechainCreation) output;
+                }
+            }
+
+            if (sidechainCreation == null)
+                throw new IllegalArgumentException("Sidechain creation transaction is not found in genesisinfo.");
```

Comment:
Instead of throwing an exception, we need to print error message and `return`.
We don't want app to crash. See the usage of `printGenesisInfoUsageMsg`

Reply:
This exception is inside try block.

Reply:
You are totally right.